src/pilot/assign_role.py

```diff
@@ -846,7 +846,30 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index,
     if role_index.index:
         role = "node:{}-{}".format(flavor, role_index.index)
 
-    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
+    # capabilities will be available if in-band introspection or
+    # out-of-band inspection is performed before
+    # assign_role.py is executed.
+    # if else statement needed to run this script before out-of-band
+    # inspection or in-band introspection has been performed.
+    if 'capabilities' in node.properties:
+        current_capabilities = node.properties.get('capabilities')
+        try:
+            cap_dict = dict(x.split(':', 1)
+                            for x in current_capabilities.split(','))
+        except ValueError:
+        # ValueError can occur if capabilities values not properly
+        # defined like capabilities='boot_mode:uefi,boot_option'.
+            raise ValueError(
```

Comment:
@gaelrehault is it necessary to raise error here or rather LOG error and return? If raising error, is it OK to raise `ValueError` or should it be different error?

Reply:
@gaelrehault, this is a friendly reminder that we would appreciate your input on this.

Reply:
If we have invalid capabilities i think it s reasonable to raise an error yes.
i have no opinion on the exception type used.